src/plugins/discover/public/application/components/discover_grid/get_render_cell_value.tsx

```diff
@@ -84,14 +84,50 @@ export const getRenderCellValueFn = (
   }
 
   if (field?.type === 'geo_point' && rowFlattened && rowFlattened[columnId]) {
-    const valueFormatted = rowFlattened[columnId] as { lat: number; lon: number };
+    const valueFormatted = rowFlattened[columnId] as {
+      lat?: number;
+      lon?: number;
+      coordinates?: number[];
+      type?: 'string';
+    };
+
+    const getLatLon = () => {
+      /**
+       * That's usually provided when using _source as source
+       */
+      if (typeof valueFormatted.lat === 'number' && typeof valueFormatted.lon === 'number') {
+        return {
+          lat: valueFormatted.lat,
+          lon: valueFormatted.lon,
+        };
+      }
+      /**
+       * That's provided when using fields as source
+       */
+      if (valueFormatted.coordinates && typeof valueFormatted.type) {
+        return {
+          lat: valueFormatted.coordinates[0],
+          lon: valueFormatted.coordinates[1],
+        };
+      }
+      /**
+       * For any other cases, to display the value
+       */
+      return { custom: valueFormatted };
+    };
+    const { lat, lon, custom } = getLatLon();
+
+    if (custom) {
+      return <div>{JSON.stringify(custom)}</div>;
+    }
+
     return (
       <div>
         {i18n.translate('discover.latitudeAndLongitude', {
```

Comment:
nit: Honestly, I don't understand why we need that logic for custom formatting `geo_point`. If we really need it what do you think about moving it into `FieldsFormats` for the corresponding type. 

To here: 
![image](https://user-images.githubusercontent.com/20072247/108034144-207cbf00-7046-11eb-9a85-c0f1b4c915da.png)

It would be nice if all plugins use **common** logic for serializing data

Reply:
I think that's actually the better approach. @kertal do you think it would be feasible to still convert that into a field formatter?

Reply:
@timroes yes, sure, we should do that. but now? or can we do it later on?

Reply:
created #91637 for that